### PR TITLE
fix: resolve sed crash in uninstall.sh due to multiple env entries

### LINKED ISSUE #4958: bug: utils/uninstall.sh crashes with sed error when .bashrc contains multiple WasmEdge env entries
### Summary

Similar to issue #4955 (where multiple installations can pollute `.bashrc` with duplicate entries), I noticed that if a user's `.bashrc` (or`.profile`) ends up with multiple identical WasmEdge `. "path/to/env"` lines, running utils/uninstall.sh completely crashes.

### Current State

In `utils/uninstall.sh` around line 263, the script uses `grep -n` to find the line number of the `env` entry, and feeds it into `sed`. If a user's file has multiple entries, `grep -n` matches multiple lines (e.g., `10\n11`), which evaluates to `sed "10\n11d"`. This throws an invalid `sed` syntax error and immediately aborts the uninstaller, leaving the system unclean.

### Expected State

The uninstaller should safely remove all WasmEdge `env` source lines from the user's profile files without crashing, regardless of how many duplicate entries exist in the file.

### Reproduction steps

1. Open your `~/.bashrc` (or `~/.profile` / `~/.bash_profile`).
2. Manually append `. "$HOME/.wasmedge/env"` at least twice at the bottom of the file.
3. Run the uninstaller script: `bash <(curl -sSf https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/uninstall.sh)`
4. Observe the `sed` syntax error and the script failing to clean up the file.

### Any logs you want to share for showing the specific issue

sed: -e expression # 1, char 5: unknown command: `\n'

### WasmEdge Version or Commit you used

master branch / 0.14.1

### Operating system information

Ubuntu 22.04

### PR DESCRIPTION
**Pull request by Sukuna0007Abhi**
## Description

Fixes #4958 

While looking at the installer side-effects discussed in #4955, I noticed a fragile edge case on the reverse side in `uninstall.sh`.

If a user ends up with duplicate WasmEdge `env` entries in their `~/.bashrc` (which easily happens if the installer is run multiple times or manually tweaked), the uninstaller completely crashes.

The script currently uses `grep -n` to grab the line number to feed to `sed`:
`line_num="$(grep -n ". \"${IPATH}/env\"" "$file" | cut -d : -f 1)"`

When there are multiple duplicate entries, `line_num` evaluates to a newline-separated list (e.g. `10\n11`). Passing this to `sed` produces an invalid command (`sed "10\n11d"`), instantly breaking the uninstallation process with a syntax error.

I swapped out the line-number deletion logic for a much safer `grep -Fv` filter. This guarantees all identical occurrences are cleanly removed without relying on `sed`'s line-deletion syntax, making the script fully robust against duplicate entries.

## Checklist

Before submitting this PR, please ensure the following:

- [x] **DCO Signed-off**: All commits are signed-off (`git commit -s`). CI workflows will only be approved if DCO check is passed.
- [x] **Commit Messages**: Run `commitlint` on your commit messages to ensure they meet the [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standards.
- [ ] **Local Tests Passed**: Provide a screenshot or logs below to prove that you have passed the test suites locally.

 *I'm exploring more and more into the WasmEdge codebase to prepare for the upcoming LFX Mentorship Term 3 — happy to adjust anything here if needed!*

### CODE REVIEW

**Comment by hydai** [comment]
Could you add a test to cover this case and ensure that the behavior matches the expected one?

**Comment by Sukuna0007Abhi** [comment]
> Could you add a test to cover this case and ensure that the behavior matches the expected one?

Thanks for looking into this, I've added a test with some dummy for this, pls review

File: utils/uninstall.sh [hunk 1]
```diff
@@ -260,12 +260,12 @@ main() {
...
-      [[ -n "$line_num" ]] || continue
-      real_file="$(resolve_path "$file")"
-      cp "$real_file" "$real_file.bak"
-      sed "${line_num}d" "$real_file.bak" > "$real_file"
-      rm -f "$real_file.bak"
+      if grep -qF ". \"${IPATH}/env\"" "$file"; then
+        real_file="$(resolve_path "$file")"
+        cp "$real_file" "$real_file.bak"
+        grep -Fv ". \"${IPATH}/env\"" "$real_file.bak" > "$real_file"
+        rm -f "$real_file.bak"
```
[the remark this answers is not in the document]
Comment (Sukuna0007Abhi) [reply utils/uninstall.sh:267]: Valid point working on it. A rare edge case found.

File: .github/workflows/test-installers.yml [hunk 2]
```diff
@@ -157,6 +157,31 @@ jobs:
...
+        # Add dummy content to verify it is preserved
+        echo "export DUMMY_VAR=1" >> ~/.bashrc
+        echo ". \"\$HOME/.wasmedge/env\"" >> ~/.bashrc
+        echo "alias dummy=ls" >> ~/.bashrc
+        echo ". \"\$HOME/.wasmedge/env\"" >> ~/.bashrc
+        
+        bash utils/uninstall.sh -q -V
+        
+        # Verify the duplicate entries are removed
+        if grep -q ". \"\$HOME/.wasmedge/env\"" ~/.bashrc; then
```
[the remark this answers is not in the document]
Comment (Sukuna0007Abhi) [reply .github/workflows/test-installers.yml]: nice catch

File: utils/uninstall.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sukuna0007Abhi) [reply utils/uninstall.sh:267]: duplicate

**Comment by Sukuna0007Abhi** [comment]
Plz review @hydai @0yi0 @SAtacker

File: .github/workflows/test-installers.yml [hunk 3]
```diff
@@ -157,6 +157,31 @@ jobs:
...
+        fi
+        
+        # Verify the expected unrelated content is preserved
+        if ! grep -qF "export DUMMY_VAR=1" ~/.bashrc || ! grep -qF "alias dummy=ls" ~/.bashrc; then
+          echo "[FAIL] Unrelated content in .bashrc was corrupted or removed"
+          exit 1
+        fi
+        
+        echo "[PASS] Duplicate entries removed and unrelated content preserved successfully"
+
```
Comment (hydai) [inline .github/workflows/test-installers.yml:184]: I have a question about this PR. This test case for the uninstaller only exists in the install.sh+linux. Could you please explain what the reason that not including install.sh+macOS and install_v2.sh+linux/macOS?

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
Hi @Sukuna0007Abhi,  
Just want to confirm if you are still going to continue with this PR or not.

**Comment by Sukuna0007Abhi** [comment]
> Hi @Sukuna0007Abhi, Just want to confirm if you are still going to continue with this PR or not.

Yes, sorry , I am working on it and will update shortly

**Comment by hydai** [comment]
Closing this PR due to inactivity. Thank you for your contribution; please feel free to reopen it if you still wish to work on it.
